### PR TITLE
Add inpage-toolbar-ui example

### PR DESCRIPTION
**Pull request by rpl**
This example shows how an add-on can integrate an UI into a webpage.

The UI resources will be packaged in the add-on itself and exposed to
the webpage by injecting an iframe which points to an "web accessible
resource" add-on URL.

This pattern is useful to any add-on which needs to integrate an UI into
the webpage (e.g. browsing enhancement toolbars, in page devtools
panels,  ad-blockers/anti-tracking in-page alerts) and it works on
Chromium-based browsers and Firefox Desktop >= 46.0 (enabled by Bug
1214658).

### CODE REVIEW

File: inpage-toolbar-ui/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# inpage-toolbar-ui
+
+## What it does ##
+
+The extension includes:
+
+* a browser action which enable/disable the in-page toolbar
```
Comment (wbamberg) [inline inpage-toolbar-ui/README.md]: -> "enables/disables"

File: inpage-toolbar-ui/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+* a content script which creates/removes the in-page toolbar iframe
+* the toolbar ui resources, packaged as web accessible resources
+
+When the user clicks the browser action button, a toolbar is shown/hidden
+in the current web page.
+
+The toolbar UI is packaged in the add-on resources, exposed to the current
+web page as web accessible resources and injected into the page by the
```
Comment (wbamberg) [inline inpage-toolbar-ui/README.md]: -> "as a web accessible resource"

File: inpage-toolbar-ui/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+content script by creating and injecting into the page an iframe which
+points to the toolbar UI page.
+
+## What it shows ##
+
+how to expose an in-page toolbar UI by creating an iframe:
```
Comment (wbamberg) [inline inpage-toolbar-ui/README.md]: -> "How"

File: inpage-toolbar-ui/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+* use web accessible resources to enable web pages to load packaged content
+* use a content script to create and inject in a web page an iframe with points
```
Comment (wbamberg) [inline inpage-toolbar-ui/README.md]: I didn't understand "with points". Should this be "which points to the packaged content" or something?

File: inpage-toolbar-ui/toolbar/ui.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// Connect to the background page.
+var port = chrome.runtime.connect();
+
+// Handle click events on the toolbar button.
+document.querySelector("#toggle").addEventListener("click", function() {
+  // Ask to the background page to toggle the toolbar on the current tab
```
Comment (wbamberg) [inline inpage-toolbar-ui/toolbar/ui.js]: -> "Ask the background page"

**Comment by wbamberg** [comment]
This looks great! Really useful example, very clean and understandable, to me anyway.

**Comment by wbamberg** [comment]
Oh, one other thing: I guess part of the point of this example is to show how, if you inject an iframe into an arbitrary web page, and the iframe points to a packaged add-on resource, then that resource gets the same access to privileged APIs that content scripts do (I think?) - meaning for instance that they can talk to the background script.

So you can make an in-content UI, and it can interact with the main extension.

I think it would be worth making this a bit more explicit, maybe just calling it out in the README "what it shows" section.

**Comment by rpl** [comment]
@wbamberg thanks a lot!
I just pushed in this PR a commit which applies the suggested fixes (and tweaks the "what it shows" section to be more explicit about the APIs available in such an iframe)

**Comment by wbamberg** [comment]
Thanks @rpl !

**Pull request merged**